server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -1590,18 +1594,18 @@ public void testExternalRefreshMetric() throws IOException {
             assertThat(shard.refreshStats().getExternalTotal(), equalTo(2L + i));
             assertThat(shard.refreshStats().getExternalTotalTimeInMillis(), greaterThanOrEqualTo(initialTotalTime));
         }
-        long externalRefreshCount = shard.refreshStats().getExternalTotal();
-
+        final long externalRefreshCount = shard.refreshStats().getExternalTotal();
+        final long extraInternalRefreshes = shard.routingEntry().primary() || shard.indexSettings().isSoftDeleteEnabled() == false ? 0 : 1;
```

Comment:
I'm not sure if this is expected or not. Seems a bit awkward. Maybe there's a better solution?